src/vm/i386/stublinkerx86.cpp

```diff
@@ -6671,6 +6671,7 @@ void FixupPrecode::ResetTargetInterlocked()
     _ASSERTE(IS_ALIGNED(this, sizeof(INT64)));
     EnsureWritableExecutablePages(this, sizeof(INT64));
     FastInterlockExchangeLong((INT64*)this, *(INT64*)&newValue);
+    FlushInstructionCache(GetCurrentProcess(), this, sizeof(INT64));
```

Comment:
`FlushInstructionCache` is no-op on x86. If you would like, you can call `ClrFlushInstructionCache`.

Reply:
Hmm on both x86 and x64 ClrFlushInstructionCache appears to just be a MemoryBarrier() ([here](https://github.com/dotnet/coreclr/blob/master/src/vm/i386/cgencpu.h#L547) and [here](https://github.com/dotnet/coreclr/blob/master/src/vm/amd64/cgencpu.h#L531)). If that's all that's necessary then a flush shouldn't be necessary at all here since there's an interlocked operation. If that's true I'll just remove the flush in Precode::SetTargetInterlocked as well.